src/common/crypto/platform/openssl/crypto.cpp

```diff
@@ -478,21 +478,48 @@ expected::ExpectedString ExtractPublicKey(const Args &args) {
 	if (ret != OPENSSL_SUCCESS) {
 		return expected::unexpected(MakeError(
 			SetupError,
-			"Failed to extract the public key from: (" + args.private_key_path
+			"Failed to extract the public key from bio: (" + args.private_key_path
 				+ "): OpenSSL BIO write failed: " + GetOpenSSLErrorMessage()));
 	}
 
-	int pending = BIO_ctrl_pending(bio_public_key.get());
+	EVP_PKEY *evp_public_key = PEM_read_bio_PUBKEY(bio_public_key.get(), NULL, NULL, NULL);
+
+	if (evp_public_key == NULL) {
```

Comment:
Like you probably can tell. This is not the proper way of doing this. However - it works - and it allows us to move on.

I will suggest to create a ticket, which digs deeper into this later, and does use the default provider for writing the public key to PKCS#8. 

Then move on with this for now

Reply:
Discussed f2f. This is fine.

Will create a task, which maybe an intern can have a look at at some point.